Chess/desk.h

```diff
@@ -1,25 +1,36 @@
 #ifndef DESK_H
 #define DESK_H
 
-#include <QObject>
+#include "common_types.h"
+#include "figure.h"
+
 #include <vector>
 
+
 class Desk : public QObject
 {
     Q_OBJECT
 public:
+    friend class Command;
+
     explicit Desk(QObject *parent = 0);

```

Comment:
Why parent is QObject? I would prefer some stronger typization like Game* of IGame*